infogami/infobase/dbstore.py

```diff
@@ -631,9 +639,7 @@ def initialize(self):
 
             self.db.update('thing', type=id, where='id=$id', vars=locals())
             self.db.insert('version', False, thing_id=id, revision=1)
-            self.db.insert(
-                'data', False, thing_id=id, revision=1, data=simplejson.dumps(data)
-            )
+            self.db.insert('data', False, thing_id=id, revision=1, data= json.dumps(data, default=self.check_datetime(t)))
```

Comment:
You only have to pass the reference to the function, so we do not need to call it. The module will call it **only** if some object is not serializable. Our function is a fallback for the JSON module.
```suggestion
            self.db.insert('data', False, thing_id=id, revision=1, data= json.dumps(data, default=self.encode_datetime))
```